src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -5585,6 +5587,28 @@ CompareTasksByTaskId(const void *leftElement, const void *rightElement)
 }
 
 
+/*
+ * CompareTasksByExecutionDuration is a helper function to compare two tasks by their execution duration.
+ */
+int
+CompareTasksByExecutionDuration(const void *leftElement, const void *rightElement)
+{
+	const Task *leftTask = *((const Task **) leftElement);
+	const Task *rightTask = *((const Task **) rightElement);
+
+	double leftTaskExecutionDuration = leftTask->fetchedExplainAnalyzeExecutionDuration;
+	double rightTaskExecutionDuration = rightTask->fetchedExplainAnalyzeExecutionDuration;
+
+	double diff = leftTaskExecutionDuration - rightTaskExecutionDuration;
```

Comment:
Can't we just do `return (int)(rightTaskExecutionDuration - leftTaskExecutionDuration)` here instead of the if checks?

Reply:
I think casting `double` to `int` may cause a miscalculation, e.g 0.345 to 0, meaning that they are equal

Reply:
What is the time unit of Duration here?

Reply:
Milliseconds

Reply:
umm that probably means that we can live with that (users will probably not care if two plans are misordered because there is 0.5 milliseconds different between them. If we still want to get 100% right order we can possibly have the following (similar to CompareTasksByTaskId) :

```
if (diff > 0) {
return -1;
}else if ( diff < 0) {
return 1;
} else {
return 0;
}
```

(@onderkalaci can probably provide input here as well :) )
Also we can append the time unit to the field such as `fetchedExplainAnalyzeExecutionDuration` -> `fetchedExplainAnalyzeExecutionDurationInMs` so that while reading the code we know the time unit :).

Reply:
-1 on returning the floating point difference
+1 on explicit if-else statements